examples/animation/animated_ui.rs

```diff
@@ -0,0 +1,184 @@
+//! Shows how to use animation clips to animate UI properties.
+
+use bevy::{
+    animation::{AnimationTarget, AnimationTargetId},
+    prelude::*,
+};
+
+// A type that represents the font size of the first text section.
+//
+// We implement `AnimatableProperty` on this.
+#[derive(Reflect)]
+struct FontSizeProperty;
+
+// A type that represents the color of the first text section.
+//
+// We implement `AnimatableProperty` on this.
+#[derive(Reflect)]
+struct TextColorProperty;
+
+// Holds information about the animation we programmatically create.
+struct AnimationInfo {
+    // The name of the animation target (in this case, the text).
+    target_name: Name,
+    // The ID of the animation target, derived from the name.
+    target_id: AnimationTargetId,
+    // The animation graph asset.
+    graph: Handle<AnimationGraph>,
+    // The index of the node within that graph.
+    node_index: AnimationNodeIndex,
+}
+
+// The entry point.
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        // Note that we don't need any systems other than the setup system,
+        // because Bevy automatically updates animations every frame.
+        .add_systems(Startup, setup)
+        .run();
+}
+
+impl AnimatableProperty for FontSizeProperty {
+    type Component = Text;
+
+    type Property = f32;
+
+    fn get_mut(component: &mut Self::Component) -> Option<&mut Self::Property> {
+        Some(&mut component.sections.get_mut(0)?.style.font_size)
+    }
+}
+
+impl AnimatableProperty for TextColorProperty {
+    type Component = Text;
+
+    type Property = Srgba;
+
+    fn get_mut(component: &mut Self::Component) -> Option<&mut Self::Property> {
+        match component.sections.get_mut(0)?.style.color {
+            Color::Srgba(ref mut color) => Some(color),
+            _ => None,
+        }
+    }
+}
+
+impl AnimationInfo {
+    // Programmatically creates the UI animation.
+    fn create(
+        animation_graphs: &mut Assets<AnimationGraph>,
+        animation_clips: &mut Assets<AnimationClip>,
+    ) -> AnimationInfo {
+        // Create an ID that identifies the text node we're going to animate.
+        let animation_target_name = Name::new("Text");
+        let animation_target_id = AnimationTargetId::from_name(&animation_target_name);
+
+        // Allocate an animation clip.
+        let mut animation_clip = AnimationClip::default();
+
+        // Create a curve that animates font size.
+        animation_clip.add_curve_to_target(
+            animation_target_id,
+            VariableCurve::linear::<AnimatablePropertyKeyframes<FontSizeProperty>>(
+                vec![0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0],
+                vec![24.0, 80.0, 24.0, 80.0, 24.0, 80.0, 24.0],
+            ),
+        );
+
+        // Create a curve that animates font color.
+        // Note that this should have the same time duration as the previous curve.
```

Comment:
```suggestion
        // Create a curve that animates font color.
        // You can construct a VariableCurve manually without the helper methods (new, linear, etc.)
        // if you prefer.
        // Note that this should have the same time duration as the previous curve.
```

Might be nice to call out explicitly that we're demonstrating different ways to construct `VariableCurve`, not that there's some hidden reason we _can't_ use `linear(...)` here as well.

Reply:
I updated the comment.